src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -0,0 +1,1317 @@
+    {
+        return ForkProject(
+            this.Solution.WithProjectFilePath(projectId, filePath),
+            translate: null,
+            forkTracker: true);
+    }
+
+    /// <inheritdoc cref="SolutionState.WithProjectCompilationOptions"/>
+    public SolutionCompilationState WithProjectCompilationOptions(
+        ProjectId projectId, CompilationOptions options)
+    {
+        return ForkProject(
+            this.Solution.WithProjectCompilationOptions(projectId, options),
+            static stateChange => new CompilationAndGeneratorDriverTranslationAction.ProjectCompilationOptionsAction(stateChange.NewProjectState, isAnalyzerConfigChange: false),
+            forkTracker: true);
+    }
+
+    /// <inheritdoc cref="SolutionState.WithProjectParseOptions"/>
+    public SolutionCompilationState WithProjectParseOptions(
+        ProjectId projectId, ParseOptions options)
+    {
+        var stateChange = this.Solution.WithProjectParseOptions(projectId, options);
+
+        if (this.PartialSemanticsEnabled)
+        {
+            // don't fork tracker with queued action since access via partial semantics can become inconsistent (throw).
+            // Since changing options is rare event, it is okay to start compilation building from scratch.
+            return ForkProject(
+                stateChange,
+                translate: null,
+                forkTracker: false);
+        }
+        else
+        {
+            return ForkProject(
+                stateChange,
+                static stateChange => new CompilationAndGeneratorDriverTranslationAction.ReplaceAllSyntaxTreesAction(stateChange.NewProjectState, isParseOptionChange: true),
+                forkTracker: true);
+        }
+    }
+
+    /// <inheritdoc cref="SolutionState.WithHasAllInformation"/>
+    public SolutionCompilationState WithHasAllInformation(
+        ProjectId projectId, bool hasAllInformation)
+    {
+        return ForkProject(
+            this.Solution.WithHasAllInformation(projectId, hasAllInformation),
+            translate: null,
+            forkTracker: true);
+    }
+
+    /// <inheritdoc cref="SolutionState.WithRunAnalyzers"/>
+    public SolutionCompilationState WithRunAnalyzers(
+        ProjectId projectId, bool runAnalyzers)
+    {
+        return ForkProject(
+            this.Solution.WithRunAnalyzers(projectId, runAnalyzers),
+            translate: null,
+            forkTracker: true);
+    }
+
+    /// <inheritdoc cref="SolutionState.WithProjectDocumentsOrder"/>
+    public SolutionCompilationState WithProjectDocumentsOrder(
+        ProjectId projectId, ImmutableList<DocumentId> documentIds)
+    {
+        return ForkProject(
+            this.Solution.WithProjectDocumentsOrder(projectId, documentIds),
+            static stateChange => new CompilationAndGeneratorDriverTranslationAction.ReplaceAllSyntaxTreesAction(stateChange.NewProjectState, isParseOptionChange: false),
+            forkTracker: true);
+    }
```

Comment:
Not a big deal but somehow this is in a different order than all the methods in SolutionState.cs.

Reply:
wdym "a different order"?